front/src/modules/trainschedule/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx

```diff
@@ -41,6 +47,69 @@ const ImportTrainScheduleConfig = ({
   const dispatch = useAppDispatch();
   const { openModal, closeModal } = useContext(ModalContext);
 
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+
+  // Function to perform the search using ciCode and chCode
+  const searchOperationalPoints = async (
+    ciCode: string,
+    chCode?: string
+  ): Promise<SearchResultItemOperationalPoint[]> => {
+    const isSearchingByTrigram = !Number.isInteger(+ciCode) && ciCode.length < 4;
+    const searchQuery = isSearchingByTrigram
+      ? ['or', ['=i', ['trigram'], ciCode], ['search', ['name'], ciCode]]
+      : ['or', ['search', ['name'], ciCode], ['like', ['to_string', ['uic']], `%${ciCode}%`]];
+
+    const payload = {
+      object: 'operationalpoint',
+      query: ['and', searchQuery],
+    };
+
+    try {
+      const results = await postSearch({ searchPayload: payload, pageSize: 101 }).unwrap();
```

Comment:
It's not super great to make an import process depend on the current infra… This makes it stateful. If the infra contains errors, it's not possible to fix it to address the issues - need to re-import the XML file as well. IOW, we loose information during the import.

Can't we generate `PathItemLocation` straight from the railml? Why do we need to search the infra?

Reply:
I'm not sure we can generate PathItemLocation, as in the XML file we only have ci / ch codes. Is it enough ? 
I rely on those codes to find associated uic or trigram.

Reply:
It seems like it's possible to convert between CI and UIC, there's just a "86" to add/remove.

One could argue that converting CI to UIC this way would make the import SNCF-specific. However I'd argue that it's already SNCF-specific by assuming a specific "ci/ch" format for the `code` XML attribute.